### DISCUSSION TITLE
[RFC] Add NamedRemainingAccounts as a BTreeMap<String, AccountInfo>

### DISCUSSION THREAD

**Discussion by kevinrodriguez-io** [otter-sec/anchor#1900] [Ideas]
Right now `RemainingAccounts` is a vector that you have to reference via positioning yourself on it, which makes further modifications and scaling harder to do. I've seen different usage patterns like commenting what they are by index, using a numeric index, but that complicates things a bit on both the client side and program side.

The request would be to make another data structure named `NamedRemainingAccounts` and deprecate `RemainingAccounts`. So in TS is a `Record<String, anchor.web3.Pubkey>` and it's properly mapped as a `BTreeMap<String, AccountInfo>` in Rust land to make it easier to grab accounts since the main usage for RemainingAccounts has been optional accounts, like receiving a mint for a treasury that can be either an ATA or Native Sol. This would work very similar to how bumps are passed, and would also keep the current data structure.

**Comment by armaniferrante** [comment]
The other use case of `remaining_accounts` I can think of is a dynamically sized Vec of accounts.
